### PR TITLE
Update Spilo version and tag

### PR DESCRIPTION
**Pull request by William-Young-97**
Updated version and tag of Spilo as requested in #977

### CODE REVIEW

**Comment by William-Young-97** [comment]
@bitsf 

Fixed the faulty tests for Spilo.

**Comment by heww** [comment]
Do you test your MR when upgrading the harbor cluster instance with the old version spilo to the new version?

As I know, spilo doesn't support upgrading from 12 t0 14 in place.

**Comment by cagiti** [comment]
It seems that this upgrade path needs to be done in 2 parts. Between spilo-12:1.6-p3 -> spilo-14:2.1-p7 are changes which support updates from POSTGRES 12 -> POSTGRES 13 -> POSTGRES 14. It means that the upgrade path would be spilo-12:1.6-p3 -> spilo-13:2.1-p1 -> spilo-14:2.1-p7 to enable operators to perform in-place upgrades of POSTGRES.

**Comment by bitsf** [comment]
run `make generated-diff-test` locally

**Comment by bitsf** [comment]
It seems CI all failed with database not ready error, not sure if you can succeed locally.
and please rebase code first then run `make generated-diff-test`

**Comment by cagiti** [comment]
> It seems CI all failed with database not ready error, not sure if you can succeed locally. and please rebase code first then run `make generated-diff-test`

All generated updates in terms of image versions have been rectified for the `manifest/clusters/deployment.yaml` when you execute `make generated-diff-test`. However there seems to be unexpected annotations to the `manifest/clusters/deployment.yaml` file

**Comment by cagiti** [comment]
@bitsf: Is it possible to get `v1.7.0` version of the `postgres-operator` built and made available [here](https://github.com/orgs/goharbor/packages/container/package/postgres-operator)

**Comment by cagiti** [comment]
@heww or @bitsf: would you be able to help with the above ☝️  at all please 🙏 ? Thanks

**Comment by bitsf** [comment]
I have transfered `ghcr.io/goharbor/spilo-14:2.1-p7`, but still failed.
image

**Comment by cagiti** [comment]
Thanks @bitsf, but I think it's failing to start the pod because it's unable to locate `v1.7.0` of the `postgres-operator` in ghcr.io/goharbor/postgres-operator. Would you be able to upload that version please? Thanks 🙏

**Comment by bitsf** [comment]
it seems the method we use kustomize to replace third party images, which in order to workaround dockerhub rate limit, is not a better idea, this introduce too much extra efforts when upgrade dependencies.

**Comment by bitsf** [comment]
you need rebase code to solve conflicts

**Comment by cagiti** [comment]
> you need rebase code to solve conflicts

Thanks, was waiting to see if the tests pass :)

File: pkg/image/components.go [hunk 1]
```diff
@@ -128,8 +128,8 @@ func init() { //nolint:gochecknoinits
 	RegisterTag("cluster-redis", "5.0-alpine", "~2.2.0", "~2.3.0", "~2.4.0", "~2.5.0", "~2.6.0")
 
 	RegisterRepository("cluster-postgresql", "registry.opensource.zalan.do/acid", "*")
-	RegisterImageName("cluster-postgresql", "spilo-12", "*")
-	RegisterTag("cluster-postgresql", "1.6-p3", "~2.2.0", "~2.3.0", "~2.4.0", "~2.5.0", "~2.6.0")
+	RegisterImageName("cluster-postgresql", "spilo-13", "*")
+	RegisterTag("cluster-postgresql", "2.1-p1", "~2.2.0", "~2.3.0", "~2.4.0", "~2.5.0")
```
Comment (bitsf) [inline pkg/image/components.go:132]: version 2.6.0 is missing

Comment (cagiti) [reply pkg/image/components.go]: it was indeed, good spot 👍

**Review by thcdrt — APPROVED** [review]

**Pull request merged**